### PR TITLE
[V2 #1660 Style Rework] Add Theme colour palette, types, and fix subsequent errors

### PR DESCRIPTION
**Pull request by W3stside**
Part of #1660 and :droplet: 's into #1664 

This PR adds our theme colour palette inside `theme > styles > colours` and the start of its types inside `theme > types` - relevant code all exported inside `theme > index.ts`

Colour palette is very minimal right now, can be iterated over as need be. Right now the `Color` type resembles:
```ts
interface Colors {
  // text
  primaryText1: Color
  secondaryText1: Color
  secondaryText2: Color
  textDisabled: Color

  // backgrounds / greys
  bg1: Color
  bg2: Color
  bgDisabled: Color

  // gradients
  gradient1: Color
  gradient2: Color

  // Base
  white: Color
  black: Color
  red1: Color
  red2: Color
  red3?: Color
  green1: Color
  green2: Color
  green3?: Color
  yellow1: Color
  yellow2: Color
  yellow3?: Color
  blue1: Color
  blue2: Color
  blue3?: Color
}
```

and the colour palettes themselves can be easily edited inside `theme > styles > colours` like:
```ts
const BASE_COLOURS = {
  // base
  white: '#FFF',
  black: '#000',
  red1: '#FF305B',
  red2: '#FF6871',
  red3: '#F82D3A',
  green1: '#00C46E',
  green2: '#09371d',
  green3: '#a9ffcd',
  yellow1: '#f1851d',
  yellow2: '#f1851d',
  blue1: '#2172E5',
  blue2: '#3F77FF',
}

export const LIGHT_COLOURS = {
  // text
  primaryText1: '#FFF',
  secondaryText1: '#EDEDED',
  secondaryText2: '#9797B8',
  textDisabled: '#31323E',

  // backgrounds / greys
  bg1: '#ffc1ff',
  bg2: '#F7F8FA',
  bgDisabled: '#ffffff80',

  // gradients
  gradient1: '#8958FF',
  gradient2: '#3F77FF',
}

export const DARK_COLOURS = {
  // text
  primaryText1: '#FFF',
  secondaryText1: '#EDEDED',
  secondaryText2: '#9797B8',
  textDisabled: '#31323E',

  // backgrounds / greys
  bg1: '#212429',
  bg2: '#2C2F36',
  bgDisabled: '#ffffff80',

  // gradients
  gradient1: '#8958FF',
  gradient2: '#3F77FF',

  // TODO: add to theme, not colour palette
  // gradientForm1: 'linear-gradient(270deg, #8958FF 0%, #3F77FF 100%)',
  // gradientForm2: 'linear-gradient(270deg, #8958FF 30%, #3F77FF 100%)',
}
```

### Testing
1. open storybook and looks at `buttons` can see the toggle changing the background colour

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
   * **🎩 [Dapp](https://pr1666--gpv1ui.review.gnosisdev.com)**: Testing web app
   * **📚 [Storybook](https://pr1666--gpv1ui.review.gnosisdev.com/storybook/)**: Component stories

File: src/components/Toggle.tsx [hunk 1]
```diff
@@ -2,8 +2,8 @@ import React from 'react'
 import styled from 'styled-components'
```
Comment (W3stside) [inline src/components/Toggle.tsx:2]: this change is because stories was throwing an error with this component

**Review by anxolin — APPROVED** [review]
Great! 🎨

I reviewed the structure, not that the actual colors make sense. Maybe @biocom has a say, or he can have a say in a PR on top of the waterfall later

File: src/components/common/Button/index.tsx [hunk 2]
```diff
@@ -1,8 +1,8 @@
 import React from 'react'
-import styled, { css, ThemeProvider } from 'styled-components'
-import { ThemeMap, ThemeValue, variants } from 'styled-theming'
+import styled, { css, /* DefaultTheme, */ ThemeProvider } from 'styled-components'
```
Comment (alfetopito) [inline src/components/common/Button/index.tsx:2]:
```suggestion
import styled, { css, ThemeProvider } from 'styled-components'
```
?

File: src/theme/styles/colours.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
+export const BASE_COLOURS = {
```
Comment (alfetopito) [inline src/theme/styles/colours.ts:1]: I only see these colors used on storybook.
Are they not used anywhere else?

**Review by alfetopito — APPROVED** [review]
Minor comments/questions

File: src/components/common/Button/index.tsx (hunk 2 above)
Comment (W3stside) [reply src/components/common/Button/index.tsx:2]: i use it later in the chain so I will ignore this for now but thanks!

File: src/theme/styles/colours.ts (hunk 3 above)
Comment (W3stside) [reply src/theme/styles/colours.ts:1]: They will be used later, yes

**Pull request merged**